extensions/iceberg/src/test/java/io/deephaven/iceberg/PyIceberg1Test.java

```diff
@@ -0,0 +1,111 @@
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class PyIceberg1Test {
+    private static final Namespace NAMESPACE = Namespace.of("dh-default");
+    private static final TableIdentifier CITIES_ID = TableIdentifier.of(NAMESPACE, "cities");
+
+    // This will need to be updated if the data is regenerated
+    private static final long SNAPSHOT_1_ID = 8833084047573120270L;
+    private static final long SNAPSHOT_2_ID = 7895378999715099054L;
+
+    private static final TableDefinition CITIES_1_TD = TableDefinition.of(
+            ColumnDefinition.ofString("city"),
+            ColumnDefinition.ofDouble("latitude"),
+            ColumnDefinition.ofDouble("lon"));
+
+    private static final TableDefinition CITIES_2_TD = TableDefinition.of(
+            ColumnDefinition.ofString("city"),
+            ColumnDefinition.ofDouble("latitude"),
+            ColumnDefinition.ofDouble("longitude"));
+
+    private IcebergCatalogAdapter catalogAdapter;
+
+    @BeforeEach
+    void setUp() throws IOException, URISyntaxException {
+        // Note: for some reason, the name must be the same name that was used to create the SqlCatalog. Potentially so
+        // a single DB file can support multiple catalogs?
+        catalogAdapter = IcebergTools.createAdapter(CatalogHelper.createJdbcCatalog(
+                "SqlCatalogName",
+                Path.of(PyIceberg1Test.class.getResource("pyiceberg-1").toURI()),
+                false));
+    }
+
+    @Test
+    void catalogInfo() {
+        assertThat(catalogAdapter.listNamespaces()).containsExactly(NAMESPACE);
+        assertThat(catalogAdapter.listTables(NAMESPACE)).containsExactly(CITIES_ID);
+        final List<Snapshot> snapshots = catalogAdapter.listSnapshots(CITIES_ID);
+        assertThat(snapshots).hasSize(2);
+        {
+            final Snapshot snapshot = snapshots.get(0);
+            assertThat(snapshot.parentId()).isNull();
+            assertThat(snapshot.schemaId()).isEqualTo(0);
+            assertThat(snapshot.sequenceNumber()).isEqualTo(1L);
+            assertThat(snapshot.snapshotId()).isEqualTo(SNAPSHOT_1_ID);
+        }
+        {
+            final Snapshot snapshot = snapshots.get(1);
+            assertThat(snapshot.parentId()).isEqualTo(SNAPSHOT_1_ID);
+            assertThat(snapshot.schemaId()).isEqualTo(1);
+            assertThat(snapshot.sequenceNumber()).isEqualTo(2L);
+            assertThat(snapshot.snapshotId()).isEqualTo(SNAPSHOT_2_ID);
+        }
+    }
+
+    @Test
+    void cities1() {
+        final Table cities1;
+        {
+            final TableDefinition td = catalogAdapter.getTableDefinition(CITIES_ID.toString(), SNAPSHOT_1_ID, null);
+            assertThat(td).isEqualTo(CITIES_1_TD);
+            cities1 = catalogAdapter.readTable(CITIES_ID, SNAPSHOT_1_ID);
```

Comment:
Note to Larry, the mismatch between signature of getTableDefinition and readTable.